snapshot-labs/sx-monorepo
# fix: add loading icon while loading sidebar starred spaces

### Summary



Closes: #232

Add a loading icon while loading sidebar starred spaces, on initial load.

![Screenshot 2024-04-08 at 13 12 22](https://github.com/snapshot-labs/sx-monorepo/assets/495709/efb250e9-126a-4353-90eb-772df1dcab0f)

### Other fixes

- Prevent duplicate data in `starredSpacesData`: previously, un-star then star the same space again will fill up `starredSpacesData` with doublons
- Avoid `getSpace()` request when the space is already cached in the spaces store: this will make the spaces starring immediate, and completely working offline.

### How to test

1. Load http://localhost:8080/#/
2. On the left sidebar, it should display a loading icon until the starred spaces are loaded
3. Un-star a space from the explore or space page
4. It should remove the space immediately from the sidebar
5. Star a space from the explore or space page
6. It should add the space immediately to the sidebar (expect a possible slight delay for the image loading from stamp), without any additional network requests.
